.github/workflows/check_interface_compatibility.yml

```diff
@@ -0,0 +1,76 @@
+name: Check interface compatibility
+
+on:
+  pull_request_target:
+
+permissions:
+  pull-requests: write
+
+jobs:
+  check-interface-compatibility:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
+        with:
+          python-version: "3.10"
+
+      - name: Extract new public interfaces
+        run: |
+          make init
+          bin/public_interface.py extract >> ${{ runner.temp }}/interfaces.new.json
+
+      - name: Backup bin/public_interface.py
+        run: |
+          # Keep a copy of bin/public_interface.py to use on the base branch
+          # So we are using the same bin/public_interface.py to process old/new codebase.
+          cp bin/public_interface.py ${{ runner.temp }}/public_interface.py
+
+      - name: Checkout the base branch
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.base_ref }}
+
+      - name: Extract original public interfaces
+        run: |
+          echo "Original HEAD: $(git rev-parse --short HEAD)"
+          make init
+          # Recover bin/public_interface.py
+          cp ${{ runner.temp }}/public_interface.py bin/public_interface.py
+          bin/public_interface.py extract >> ${{ runner.temp }}/interfaces.original.json
+
+      - name: Archive interface files for debugging
```

Comment:
Why do we need this? Isn't the script output good enough?

Reply:
This is still draft and you are reviewing the stuff used for my debugging.